src/characters/alice.ts

```diff
@@ -0,0 +1,196 @@
+import { Provider } from '@ethersproject/providers';
+import { PublicKey, Signer, VerifiedKeyFrag } from 'umbral-pre';
+
+import { StakingEscrowAgent } from '../agents/staking-escrow';
+import { NucypherKeyring } from '../crypto/keyring';
+import {
+  DelegatingPower,
+  SigningPower,
+  TransactingPower,
+} from '../crypto/powers';
+import { MessageKit } from '../kits/message';
+import {
+  BlockchainPolicy,
+  BlockchainPolicyParameters,
+  EnactedPolicy,
+} from '../policies/policy';
+import { ChecksumAddress, Configuration } from '../types';
+import {
+  calculatePeriodDuration,
+  dateAtPeriod,
+  mergeWithoutUndefined,
+} from '../utils';
+
+import { Bob } from './bob';
+import { Porter } from './porter';
+
+export class Alice {
+  private readonly porter: Porter;
+  public readonly transactingPower: TransactingPower;
+  private readonly delegatingPower: DelegatingPower;
+  private readonly signingPower: SigningPower;
+
+  private constructor(
+    config: Configuration,
+    signingPower: SigningPower,
+    delegatingPower: DelegatingPower,
+    transactingPower: TransactingPower
+  ) {
+    this.porter = new Porter(config.porterUri);
+    this.signingPower = signingPower;
+    this.delegatingPower = delegatingPower;
+    this.transactingPower = transactingPower;
+  }
+
+  public get verifyingKey(): PublicKey {
+    return this.signingPower.publicKey;
+  }
+
+  public get signer(): Signer {
+    return this.signingPower.signer;
+  }
+
+  public static fromSecretKey(
+    config: Configuration,
+    secretKey: Uint8Array
```

Comment:
Judging by how it is used in `NucypherKeyring`, it is rather secret material than a secret key (as in `umbral.SecretKey`), right?